filedescriptor/src/unix.rs

```diff
@@ -217,6 +265,41 @@ impl FileDescriptor {
         }
         Ok(())
     }
+
+    /// Attempt to duplicate the underlying handle from an object that is
+    /// representable as the system `RawFileDescriptor` type and assign it to
+    /// a destination file descriptor. It then returns a `FileDescriptor`
+    /// wrapped around the duplicate.  Since the duplication requires kernel
+    /// resources that may not be available, this is a potentially fallible operation.
+    /// The returned handle has a separate lifetime from the source, but
+    /// references the same object at the kernel level.
+    pub unsafe fn dup2<F: AsRawFileDescriptor>(f: &F, dest_fd: RawFd) -> anyhow::Result<Self> {
+        OwnedHandle::dup2_impl(f, dest_fd).map(|handle| Self { handle })
+    }
+
+    pub(crate) fn redirect_stdin<F: AsRawFileDescriptor>(f: &F) -> anyhow::Result<FileDescriptor> {
+        let std_original = FileDescriptor::dup(libc::STDIN_FILENO)?;
+        let cloned_handle = OwnedHandle::dup(f)?;
+        unsafe { FileDescriptor::dup2(cloned_handle.handle, libc::STDIN_FILENO) }?;
```

Comment:
I was envisioning passing `f` directly to `dup2` and avoiding the intermediate `cloned_handle`:

```suggestion
        unsafe { FileDescriptor::dup2(f, libc::STDIN_FILENO) }?;
```

Reply:
Done